R/CreatePackageVignette.R

```diff
@@ -0,0 +1,135 @@
+#' @title pkgnet Report as Vignette
+#' @name CreatePackageVignette
+#' @description Create pkgnet package report as an R Markdown vignette. This
+#'    vignette is able to be rendered into a standard HTML vignette with
+#'    the \code{\link[knitr:vignette_engines]{knitr::rmarkdown}} vignette engine
+#'    into HTML vignettes upon package building. It is also compatible with
+#'    \code{\link[pkgdown:build_articles]{pkgdown}} sites. See the vignette
+#'    \href{https://cran.r-project.org/web/packages/pkgnet/vignettes/publishing-reports.html}{
+#'    "Publishing Your pkgnet Package Report"} for details about how to use this
+#'    function, as well as
+#'    \href{https://cran.r-project.org/web/packages/pkgnet/vignettes/pkgnet-report.html}{
+#'    our example for pkgnet}.
+#' @param pkg_name (string) name of a package
+#' @param pkg_reporters (list) a list of initialized package reporters
+#' @param vignette_path (string) The path and filename of the output vignette
+#'    file. The default assumes your working directory is the package root.
+#' @importFrom rlang enexpr
+#' @importFrom assertthat assert_that is.string is.readable
+#' @importFrom tools file_ext
+#' @importFrom R6 is.R6Class
+#' @importFrom glue glue
+#' @export
+CreatePackageVignette <- function(pkg_name
```

Comment:
alright I've been thinking about this...I have an idea we could use to simplify this a bit.

Since the intended use is "run this interactively when you already have the source of the package sitting on your filesystem", I think we should use the same interface as `devtools::test()`, `devtools::document()`, and `devtools::check()` (tools that people are VERY familiar with already). This would look like:

```
CreatePackageVignette(
    pkg = ".",
    pkg_reporters = whatever,
    vignette_path = whatever
)
```

Benefits this would have:

1. No new complexity (since we are already parsing the DESCRIPTION file with `read.dcf()`, we can just get `pkg_name` from there)
2. Consistency with existing tools in the R developer ecosystem
3. Default call becomes `CreatePackageVignette()` with no args and "just works"
4. Cut some of the "try to figure out what folder you're in" complexity from the internals

@bburns632 @jayqi what do you think?

Reply:
I agree that would help simplify the interface. A few potential hiccups I see are:

1. Do we force the vignette report to always include pkg_path and calculate test coverage?
2. What if a user did want to create this vignette format without having the source code? Having the source code is not strictly necessarily under my original interface.
3. Do we worry about the source version diverging from the loaded version?  I believe devtools loads the source code for those functions. Do we need to load it too to ensure that there is consistency? FunctionReporter generally has this problem too in other usages, but I feel like we will fail the principle of least astonishment if we _don't_ load the source code, if the main entrypoint is that a user is pointing the function at their package root.

Reply:
I don't understand how `#1` is related to my recommendation.

I'm struggling to come up with a realistic use for "I want to make a vignette but I don't have the package source code". Why would someone who doesn't have the package source code care about making an R vignette instead of just running `CreatePackageReport()`?

`3` is something I hadn't thought of, fair point. I think as long as we document the fact that you have to do `devtools::install()` or `install.packages()` or `R CMD INSTALL` before running this, it's ok. You have to have installed the package already in your current interface, so while I get that it's a concern I don't think it's one that informs whether or not to go with my suggestion.

In every R package I've ever written, I always have an order of steps like this in CI or publishing:

1. `devtools::install()`
2. `devtools::document()`
3. `R CMD CHECK` (also running unit tests`

So I think the "you have to install first" is a tolerable divergence from what `devtools::test()` etc. do anyway.

I also want to reiterate that the "simplify the interface" benefit of what I'm proposing is nice, but my main reason for proposing this is the "consistency with all the devtools stuff" reason. I think that is a nice way for us to integrate with normal R developer workflows.

Reply:
Okay @jameslamb, I think what you said all makes sense. I agree that overall the balance is we benefit from having it work with a devtools-style interface.

Is it confusing if this interface is different than `CreatePackageReport`? Should we name this function something else to make it clear not to expect a similar interface?

Reply:
I don't think it's confusing to have a different interface for this function

Reply:
This whole function is tailored for a very specific scenario.  I would not think the above is all that confusing, especially since I imagine 90% of the uses will be without parameters.

Reply:
^ agree with @bburns632